src/components/swap/SwapBuyFiatButton.tsx

```diff
@@ -0,0 +1,141 @@
+import { Trans } from '@lingui/macro'
+import { ButtonText } from 'components/Button'
+import { MouseoverTooltipContent } from 'components/Tooltip'
+import { useWalletDrawer } from 'components/WalletDropdown'
+import { useCallback, useEffect, useState } from 'react'
+import { useBuyFiatFlowCompleted } from 'state/user/hooks'
+import styled from 'styled-components/macro'
+import { ExternalLink } from 'theme'
+
+import { useFiatOnrampAvailability, useOpenModal } from '../../state/application/hooks'
+import { ApplicationModal } from '../../state/application/reducer'
+
+const Dot = styled.div`
+  height: 8px;
+  width: 8px;
+  background-color: ${({ theme }) => theme.accentActive};
+  border-radius: 50%;
+`
+
+export const MOONPAY_REGION_AVAILABILITY_ARTICLE =
+  'https://support.uniswap.org/hc/en-us/articles/11306664890381-Why-isn-t-MoonPay-available-in-my-region-'
+
+enum BuyFiatFlowState {
+  // Default initial state. User is not actively trying to buy fiat.
+  INACTIVE,
+  // Buy fiat flow is active and region availability has been checked.
+  ACTIVE_POST_REGION_CHECK,
+  // Buy fiat flow is active, feature is available in user's region & needs wallet connection.
+  ACTIVE_NEEDS_WALLET_CONNECTION,
+}
+
+const StyledTextButton = styled(ButtonText)`
+  color: ${({ theme }) => theme.textSecondary};
+  gap: 4px;
+  &:focus {
+    text-decoration: none;
+  }
+  &:active {
+    text-decoration: none;
+  }
+`
+
+const POPOVER_DELAY_BEFORE_SHOW_MS = 500
+
+export default function SwapBuyFiatButton() {
+  // Importing useWeb3React here is necessary for mocking different behavior in different tests in SwapBuyFiatButton.test.tsx
+
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const { useWeb3React } = require('@web3-react/core')
+  const { account } = useWeb3React()
+  const openFiatOnRampModal = useOpenModal(ApplicationModal.FIAT_ONRAMP)
+  const [buyFiatFlowCompleted, setBuyFiatFlowCompleted] = useBuyFiatFlowCompleted()
+  const [checkFiatRegionAvailability, setCheckFiatRegionAvailability] = useState(false)
```

Comment:
`is`/`should` in the boolean variables gating behavior makes this easier to follow:

```suggestion
  const [isBuyFiatFlowCompleted, setBuyFiatFlowCompleted] = useBuyFiatFlowCompleted()
  const [shouldCheckFiatRegionAvailability, setCheckFiatRegionAvailability] = useState(false)
```

Also below, `isWalletDrawerOpen`.

Reply:
react useState convention would necessitate changing the set functions to "setIsBuyFiatFlowCompleted" and "setShouldCheckFiatRegionAvailability", which seem overly wordy and confusing to me. also the use of "is" for booleans is highly inconsistent across the codebase so I dont really agree with needing this change.